scripts/autogen

```diff
@@ -772,20 +771,66 @@ def check_asm_register_aliases():
         check_asm_register_aliases_for_file(asm_file)
 
 
-def gen_hol_light_asm_file(infile, outfile, dry_run=False):
+def update_via_simpasm(
+    infile_full, outdir, outfile=None, cflags=None, preserve_header=True, dry_run=False
+):
+
+    _, infile = os.path.split(infile_full)
+    if outfile is None:
+        outfile = infile
+    outfile_full = os.path.join(outdir, outfile)
+
+    # Check if we need to use a cross-compiler
+    if "aarch64" in infile_full:
+        source_arch = "aarch64"
+    elif "x86_64" in infile_full:
+        source_arch = "x86_64"
+    else:
+        raise Exception(f"Could not detect architecture of source file {infile_full}.")
+    # Check native architecture
+    if platform.machine().lower() in ["arm64", "aarch64"]:
+        native_arch = "aarch64"
+    else:
+        native_arch = "x86_64"
+
+    if native_arch != source_arch:
+        if platform.system() == "Darwin":
+            print(
+                f"WARNING: Skipping simplification of file {infile} -- "
```

Comment:
Discussions with @potsrevennil yesterday suggest that this limitation can be easily overcome. Please open an issue referencing this line so we do not forget to address this.

Reply:
Changed the behaviour here to print the warning whenever a cross-chain is unavailable, not specifically when it's Darwin.